### PR TITLE
feat(examples): Introduce ExpressJS on Node

### PR DESCRIPTION
**Pull request by razvand**
Introduce ExpressJS (Node framework) to be deployed on KraftCloud. It uses binary compatibility mode. It uses the `node:21` runtime.

Add:

* `Kraftfile`: build / run rules, including pulling the `node` image
* `Dockerfile`: ExpressJS filesystem
* `README.md`: document how to use
* `package.json`: install dependencies
* `app/index.js`: the actual ExpressJS implementation

### CODE REVIEW

File: node21-expressjs/README.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Node 21 ExpressJS Example
+
+This example uses the [ExpressJS Node framework](https://expressjs.com/).
+To run this example, first [install the `kraft` CLI tool](https://unikraft.org/docs/cli).
+
+Then, clone this repository and `cd` into this directory.
+To deploy this application on KraftCloud, invoke:
+
+```
+kraft cloud deploy -p 443:8080 -M 256 .
```
Comment (craciunoiuc) [inline node21-expressjs/README.md:10]: Is `256` enough? for the test `2048` is used

File: node21-expressjs/app/index.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+const express = require('express')
+const app = express()
+const port = 3000
```
Comment (craciunoiuc) [inline node21-expressjs/app/index.js:3]: `3000` or `8080`?

**Review by craciunoiuc — CHANGES_REQUESTED** [review]
two comments from me

File: node21-expressjs/README.md (hunk 1 above)
Comment (craciunoiuc) [reply node21-expressjs/README.md:10]: actually `512` is used in the test

File: node21-expressjs/app/index.js (hunk 2 above)
Comment (craciunoiuc) [reply node21-expressjs/app/index.js:3]: I see in the test you used 3000, so I guess one for sure needs changing

Comment (razvand) [reply node21-expressjs/app/index.js:3]: @craciunoiuc , it's `3000`.

File: node21-expressjs/README.md (hunk 1 above)
Comment (razvand) [reply node21-expressjs/README.md:10]: Yes, `256` is enough. I updated the [workflow PR](https://github.com/kraftcloud/catalog/pull/128) to use `256`.

**Pull request merged**